lgc/interface/lgc/Pipeline.h

```diff
@@ -551,6 +551,10 @@ class Pipeline {
   // Set the shader stage mask
   virtual void setShaderStageMask(unsigned mask) = 0;
 
+  // Set whether pre-rasterization part has a geometry shader
+  // NOTE: Only applicable in the part pipeline compilation mode.
```

Comment:
The difference with "part pipeline compilation" is that you have all the shaders and pipeline state available, you're just choosing to compile the two parts separately. Thus the front-end can tell the LGC compilation of the fragment shader that a GS is present. For general unlinked FS compilation, if the FS does something that relies on knowing whether a GS is present (has 16-bit inputs), then it would have to be one of those cases that fails and falls back to full pipeline compilation.

Reply:
The difference between "part pipeline compilation" and relocatable shader compilation is more just knowledge of the FS inputs.  We should talk about the differences because they could share more code that you think.

In general, LGC should be agnostic to whether or not part pipeline compilation is used or relocatable.  In both cases, it is doing an unlinked compilation.  If the bit is set, then it should be used.

If relocatable shader compilation decides to set this option, which is has to info to do, then it should work.

Reply:
This interface change has a very specific use - to do better in/out packing, see the self-contained commit "Part pipeline: Fix for dword packing". Since relocatable shader path does not do the packing, there is no need for relocatable compilation to set this option.